lib/astutils.cpp

```diff
@@ -3647,3 +3647,82 @@ bool isUnevaluated(const Token *tok)
 {
     return Token::Match(tok, "alignof|_Alignof|_alignof|__alignof|__alignof__|decltype|offsetof|sizeof|typeid|typeof|__typeof__ (");
 }
+
+static std::set<std::string> getSwitchExprs(const Token *startbrace, bool &hasDefault)
+{
+    std::set<std::string> exprs;
+    const Token *endbrace = startbrace->link();
+    if (!endbrace)
+        return {};
+
+    hasDefault = false;
+    for (const Token *tok = startbrace->next(); tok && tok != endbrace; tok = tok->next()) {
+        if (Token::simpleMatch(tok, "{")) {
+            tok = tok->link();
```

Comment:
technically I think it would be better to only skip inner switch scopes. This is valid but weird code :
```
#define VALUE 0

int dostuff(void);

int foo(int x) {
    switch(x) {
    case VALUE+1:
        if (dostuff()) {
            case VALUE+2:
               return 1;
        }
        return 2;
    }
}
```